Tried to get compatibility with workchains, this version fails because of pw2wannier in strain calc.

DanielMarchand opened the pull request.

Commit:
this FAILS

Commit:
restored old example default

greschd commented:
The root cause of our problems here is that the `structure` is passed into the SCF / NSCF / BANDS steps _explicitly_, instead of via the top-level input of the workchain.
As a result, the "normal" optimization workchain works fine because the tests pass the same structure in both places. In the strained ones, the structure is changed before handing it off to the optimization workchain. The PW calculations still get the unstrained ones (directly), so that fails.

Mostly, we need to change how the structure is passed into these sub-workchains, and then there are some minor changes to the tests to be made - I have a fix mostly cooked up.

Commit:
Fix passing of the input structure to QE sub-workchains

When replacing the bare QE calculations with workchains, their
input was changed s.t. the structure needed to be passed to each
part. This breaks the strained workflow, because it modifies the
structure before passing it on.
In this fix, we explicitly pass on the structure to the QE sub-
workflows.
